python/llm/src/bigdl/llm/transformers/models/chatglm.py

```diff
@@ -67,6 +67,8 @@ def attention_fn(
         cache_v = cache_v.permute(1, 2, 0, 3)
         past_length = cache_k.size(2)
         if cache_k.stride()[1] <= cache_k.size(2) * cache_k.size(3):
+            if device == torch.device('xpu:0'):
```

Comment:
how about other variations (e.g., `xpu:1`)?

Reply:
And what if there are multiple GPU's in the system?

Reply:
Have revised to judge device type.

Reply:
> Have revised to judge device type.

If there are multiple GPU's on the same system, will this create correctness problem?

Reply:
In my understanding, this function just clear unused cache, so it may not cause correctness problem, but I am not very sure.

Reply:
> In my understanding, this function just clear unused cache

Then it's probably fine